### PR TITLE
Bugfix/select

### PR DESCRIPTION
**Pull request by Hurenka**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing applications:)

**What changes did you make?** (Give an overview)
I've created a new Select component and replaced the old one everywhere.

**Is there anything you'd like reviewers to focus on?**
This has to be carefully tested, in order to check, that I didn't missed any functionality. UI/UX suggestions are also welcomed.

**How Has This Been Tested?** (put an "X" next to an item)

- [X] No need to
- [X] Manually(please, describe, when necessary)
- [X] Unit checks
- [X] Integration checks
- [X] Covered by existing automation

**Checklist** (put an "X" next to an item, otherwise PR will fail)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [X] My changes generate no new warnings(e.g. Sonar is happy)
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

### CODE REVIEW

File: kafka-ui-react-app/src/components/ConsumerGroups/Details/ResetOffsets/__test__/ResetOffsets.spec.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import fetchMock from 'fetch-mock';
 import { Route } from 'react-router';
-import { screen, waitFor, fireEvent } from '@testing-library/react';
+import { screen, waitFor, fireEvent, within } from '@testing-library/react';
```
Comment (NelyDavtyan) [inline kafka-ui-react-app/src/components/ConsumerGroups/Details/ResetOffsets/__test__/ResetOffsets.spec.tsx:4]: 'within' is defined but never used
delete it if you are not using it

File: kafka-ui-react-app/src/components/Schemas/List/GlobalSchemaSelector/GlobalSchemaSelector.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import ConfirmationModal from 'components/common/ConfirmationModal/ConfirmationModal';
-import Select from 'components/common/Select/Select';
+import Select, { SelectOption } from 'components/common/Select/Select';
```
Comment (NelyDavtyan) [inline kafka-ui-react-app/src/components/Schemas/List/GlobalSchemaSelector/GlobalSchemaSelector.tsx:2]: 'SelectOption' is defined but never used
delete if you are not using it

**Review by workshur — APPROVED** [review]
LGTM! Please deal with eslint warnings and merge it

File: kafka-ui-react-app/src/components/Schemas/List/GlobalSchemaSelector/__test__/GlobalSchemaSelector.spec.tsx [hunk 3]
```diff
@@ -47,22 +52,22 @@ describe('GlobalSchemaSelector', () => {
...
+  //   selectForwardOption();
+  //   expect(screen.getByText('Confirm the action')).toBeInTheDocument();
+  // });
+  //
+  // it('resets select value when cancel is clicked', () => {
+  //   selectForwardOption();
+  //   userEvent.click(screen.getByText('Cancel'));
+  //   expect(screen.queryByText('Confirm the action')).not.toBeInTheDocument();
+  //   expectOptionIsSelected(CompatibilityLevelCompatibilityEnum.FULL);
+  // });
```
Comment (Motherships) [inline kafka-ui-react-app/src/components/Schemas/List/GlobalSchemaSelector/__test__/GlobalSchemaSelector.spec.tsx:70]: Sonar will be pleased if we remove this
```suggestion
```

File: kafka-ui-react-app/src/components/Topics/Topic/Details/Messages/Filters/Filters.tsx [hunk 4]
```diff
@@ -76,7 +85,7 @@ const Filters: React.FC<FiltersProps> = ({
   );
 
   const [attempt, setAttempt] = React.useState(0);
-  const [seekType, setSeekType] = React.useState<SeekType>(
+  const [currentSeekType, setSeekType] = React.useState<SeekType>(
```
Comment (Motherships) [inline kafka-ui-react-app/src/components/Topics/Topic/Details/Messages/Filters/Filters.tsx:88]: It's either 
```js
const [currentSeekType, setCurrentSeekType] = React.useState<SeekType>(
```
or 
```js
  const [seekType, setSeekType] = React.useState<SeekType>(
```

File: kafka-ui-react-app/src/components/common/Select/Select.tsx [hunk 5]
```diff
@@ -1,56 +1,131 @@
...
+    setSelectedOption(option.value);
+    if (onChange) onChange(option.value);
+    setShowOptions(false);
+  };
+
   return (
-    <div className={`select-wrapper ${className}`}>
+    <div ref={selectContainerRef}>
       {isLive && <LiveIcon />}
       {name ? (
```
Comment (Motherships) [inline kafka-ui-react-app/src/components/common/Select/Select.tsx:62]: I have a hard time understanding this condition
Both branches of condition seem identical

**Review by Motherships — APPROVED** [review]
I don't really know how to rewrite line which sonar doesn't like to make code better so LGTM

**Review by NelyDavtyan — APPROVED** [review]

**Pull request merged**